aya/src/util.rs

```diff
@@ -95,6 +95,39 @@ fn parse_kernel_symbols(reader: impl BufRead) -> Result<BTreeMap<u64, String>, i
     Ok(syms)
 }
 
+/// Returns the prefix used by syscalls.
+///
+/// # Example
+///
+/// ```no_run
+/// use aya::util::syscall_prefix;
+/// let prefix = syscall_prefix()?;
+/// let syscall_fname = format!("{prefix}exec");
+/// ```
+///
+/// # Errors
+///
+/// Returns [`std::io::ErrorKind::NotFound`] if the prefix can't be guessed. Returns other [`std::io::Error`] kinds if `/proc/kallsyms` can't be opened or is somehow invalid.
+pub fn syscall_prefix() -> Result<&'static str, io::Error> {
+    const PREFIXES: [&str; 7] = [
+        "sys_",
+        "__x64_sys_",
+        "__x32_compat_sys_",
+        "__ia32_compat_sys_",
+        "__arm64_sys_",
+        "__s390x_sys_",
+        "__s390_sys_",
+    ];
+    let ksym = kernel_symbols()?;
+    let values = ksym.into_values().collect::<Vec<_>>();
```

Comment:
I don't think we should collect here, you can use `ksym.values()` multiple times

Reply:
because there is no contains in values(), I've replaced contains() with `any()`